draft-ietf-quic-transport.md

```diff
@@ -2344,54 +2382,50 @@ senders from exceeding a receiver's buffer capacity for the connection, and (ii)
-sending a WINDOW_UPDATE frame.  A receiver MUST NOT renege on an advertisement;
-that is, once a receiver advertises an offset via a WINDOW_UPDATE frame, it MUST
-NOT subsequently advertise a smaller offset.  A sender may receive WINDOW_UPDATE
-frames out of order; a sender MUST therefore ignore any WINDOW_UPDATE that
-does not move the window forward.
+A receiver MAY advertise a larger offset at any point by sending MAX_DATA or
+MAX_STREAM_DATA frames.  A receiver MUST NOT renege on an advertisement; that
+is, once a receiver advertises an offset, it MUST NOT subsequently advertise a
+smaller offset.  A sender could receive MAX_DATA or MAX_STREAM_DATA frames out
+of order; a sender MUST therefore ignore any flow control offset that does not
+move the window forward.
 
 A receiver MUST close the connection with a
 QUIC_FLOW_CONTROL_RECEIVED_TOO_MUCH_DATA error ({{error-handling}}) if the
-peer violates the advertised stream or connection flow control windows.
+peer violates the advertised connection or stream data limits.
 
 A sender MUST send BLOCKED frames to indicate it has data to write but is
 blocked by lack of connection or stream flow control credit.  BLOCKED frames are
 expected to be sent infrequently in common cases, but they are considered useful
 for debugging and monitoring purposes.
 
-A receiver advertises credit for a stream by sending a WINDOW_UPDATE frame with
-the Stream ID set appropriately. A receiver may use the current offset of data
-consumed to determine the flow control offset to be advertised.
-A receiver MAY send copies of a WINDOW_UPDATE frame in multiple packets in order
-to make sure that the sender receives it before running out of flow control
-credit, even if one of the packets is lost.
+A receiver advertises credit for a stream by sending a MAX_STREAM_DATA frame
+with the Stream ID set appropriately. A receiver could use the current offset of
+data consumed to determine the flow control offset to be advertised.  A receiver
+MAY send MAX_STREAM_DATA frames in multiple packets in order to make sure that
+the sender receives an update before running out of flow control credit, even if
+one of the packets is lost.
 
 Connection flow control is a limit to the total bytes of stream data sent in
-STREAM frames on all streams contributing to connection flow control.  A
-receiver advertises credit for a connection by sending a WINDOW_UPDATE frame
-with the Stream ID set to zero (0x00).  A receiver maintains a cumulative sum of
-bytes received on all streams contributing to connection-level flow control, to
-check for flow control violations. A receiver may maintain a cumulative sum of
-bytes consumed on all contributing streams to determine the connection-level
-flow control offset to be advertised.
+STREAM frames on all streams.  A receiver advertises credit for a connection by
+sending a MAX_DATA frame.  A receiver maintains a cumulative sum of bytes
```

Comment:
It is not clear if stream id 0 counts against this, even if it is not punished for violating the connection max_data limit.